tests/test_parser.py

```diff
@@ -74,13 +74,16 @@ def test_expression(self):
         warn.expression(exp.Hint, y="")
         assert isinstance(warn.errors[0], ParseError)
 
-    def test_function_arguments_validation(self):
+    def test_parse_errors(self):
         with self.assertRaises(ParseError):
             parse_one("IF(a > 0, a, b, c)")
 
         with self.assertRaises(ParseError):
             parse_one("IF(a > 0)")
 
+        with self.assertRaises(ParseError):
+            parse_one("WITH cte AS (SELECT * FROM x)")
```

Comment:
just double checking, we can still do a partial parse?

Reply:
Well, there is not CTE expression type anymore, so we'd have to arbitrarily pick an expression type to assume